src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -193,6 +192,50 @@ bool IHttpBodyControlFeature.AllowSynchronousIO
             }
         }
 
+        PipeWriter IResponseBodyPipeFeature.ResponseBodyPipe
+        {
+            get
+            {
+                return ResponsePipeWriter;
+            }
+            set
+            {
+                ResponsePipeWriter = value;
+
+                if (!object.ReferenceEquals(_cachedResponsePipeWriter, ResponsePipeWriter))
```

Comment:
I wouldn't even bother doing this ReferenceEquals check. Just create the WriteOnlyPipeStream no matter what.

If you really want to be more efficient in the unlikely case someone sets ResponsePipeWriter to its current value, you could put the following at the top of the setter for the same effect:

```c#
if (object.ReferenceEquals(value, ResponsePipeWriter) return;
```

But I don't think it's worth even wasting one line of code optimizing this scenario.

Reply:
Yeah, I kind of realized this optimization is so irrelevant that we should just ignore it.